commit 7672c24fc727a3241f4cb013a0a34a10b40e52c9 (refs/changes/15/4015/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-05-28T15:18:33-07:00 (1 year, 4 months ago)
    
    TRITON-438 TRITON-85 tweak to Amon log-scan event message broke test suite

diff --git a/master/lib/notifications/webhook.js b/master/lib/notifications/webhook.js
index 6534a05..9e2f6e7 100644
--- a/master/lib/notifications/webhook.js
+++ b/master/lib/notifications/webhook.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /**
@@ -69,8 +69,7 @@ Webhook.prototype.notify = function (options, callback) {
     var address = options.contact.address;
     var event = options.event;
     var log = this.log.child({event: event.uuid}, true);
-    log.info({address: address, user: user.uuid, alarm: alarm.id},
-        'email notify');
+    log.info({address: address, user: user.uuid, alarm: alarm.id}, 'notify');
 
     var url = urlParse(address);
     var reqOptions = {
diff --git a/plugins/lib/log-scan.js b/plugins/lib/log-scan.js
index 62d8727..301be4c 100644
--- a/plugins/lib/log-scan.js
+++ b/plugins/lib/log-scan.js
@@ -126,6 +126,9 @@ LogScanProbe.prototype._getMessage = function () {
         } else {
             msg = 'Log "' + this._pathsCache[0] + '" matched';
         }
+        if (this.matcher) {
+            msg += ' ' + this.matcher.toString();
+        }
         if (this.threshold > 1) {
             msg += format(' >=%d times in %d seconds.',
                 this.threshold, this.period);
diff --git a/test/probegroups.test.js b/test/probegroups.test.js
index a393620..392513b 100644
--- a/test/probegroups.test.js
+++ b/test/probegroups.test.js
@@ -248,9 +248,11 @@ test('got webhook notification for valhalla fault', function (t) {
                 var notification = webhooks[0].body;
                 t.ok(notification.alarm, 'notification.alarm');
                 t.ok(notification.message.indexOf('battle.log') !== -1,
-                    'message mentions "battle.log"');
+                    'message mentions "battle.log": '
+                        + JSON.stringify(notification.message));
                 t.ok(notification.message.indexOf('a warrior died') !== -1,
-                    'message mentions "a warrior died"');
+                    'message mentions "a warrior died": '
+                        + JSON.stringify(notification.message));
                 t.ok(notification.event, 'notification.event');
                 t.equal(notification.event.machine, prep.headnodeUuid,
                     'event was for the headnode');
@@ -324,9 +326,11 @@ test('got second webhook notification for valhalla fault', function (t) {
                 var notification = webhooks[0].body;
                 t.ok(notification.alarm, 'notification.alarm');
                 t.ok(notification.message.indexOf('battle.log') !== -1,
-                    'message mentions "battle.log"');
+                    'message mentions "battle.log": '
+                        + JSON.stringify(notification.message));
                 t.ok(notification.message.indexOf('another warrior') !== -1,
-                    'message mentions "another warrior"');
+                    'message mentions "another warrior": '
+                        + JSON.stringify(notification.message));
                 t.ok(notification.event, 'notification.event');
                 t.equal(notification.event.machine, prep.headnodeUuid,
                     'event was for the headnode');
